File: server/lib/CouplingDataService.ts
import * as monk from "monk";

export default class CouplingDataService {

    public database;
    playersCollection;
    historyCollection;
    tribesCollection;
    private pinCollection;

    constructor(public mongoUrl) {
        this.database = monk.default(mongoUrl);
        this.playersCollection = this.database.get('players');
        this.historyCollection = this.database.get('history');
        this.tribesCollection = this.database.get('tribes');
        this.pinCollection = this.database.get('pins');
    }

}

File: server/src/main/kotlin/com/zegreatrob/coupling/server/CouplingDataService.kt
package com.zegreatrob.coupling.server

import kotlin.js.Json
import kotlin.js.json

fun couplingDataService(mongoUrl: String): Json {
    val database = com.zegreatrob.coupling.mongo.external.monk.default(mongoUrl)
    return json(
        "database" to database,
        "playersCollection" to database.get("players"),
        "historyCollection" to database.get("history"),
        "tribesCollection" to database.get("tribes"),
        "pinCollection" to database.get("pins"),
        "usersCollection" to database.get("users")
    )
}